src/xo-app/settings/acls/index.js

```diff
@@ -1,39 +1,133 @@
+import _ from 'messages'
 import ActionButton from 'action-button'
+import ActionRowButton from 'action-row-button'
+import Component from 'base-component'
 import forEach from 'lodash/forEach'
 import isEmpty from 'lodash/isEmpty'
 import keyBy from 'lodash/keyBy'
 import map from 'lodash/map'
-import orderBy from 'lodash/orderBy'
+import pickBy from 'lodash/pickBy'
+import React from 'react'
+import renderXoItem, { renderXoItemFromId } from 'render-xo-item'
+import SortedTable from 'sorted-table'
 import Upgrade from 'xoa-upgrade'
-import React, { Component } from 'react'
-import { addSubscriptions } from 'utils'
+import { connectStore } from 'utils'
 import { Container } from 'grid'
 import { error } from 'notification'
-import { renderXoItemFromId } from 'render-xo-item'
 import { SelectHighLevelObjects, SelectRole, SelectSubject } from 'select-objects'
+
+import {
+  createGetObjectsOfType,
+  createSelector
+} from 'selectors'
+
 import {
   addAcl,
+  editAcl,
   removeAcl,
   subscribeAcls,
   subscribeGroups,
+  subscribeRoles,
   subscribeUsers
 } from 'xo'
 
-@addSubscriptions({
-  groups: cb => subscribeGroups(groups => cb(keyBy(groups, 'id'))),
-  users: cb => subscribeUsers(users => cb(keyBy(users, 'id')))
+const handleRoleChange = (role, {subject, object, action}) => {
```

Comment:
What do you mean ? Inlining it where it is used, as a prop for a Component ? Won't this be hard to indent/read ?...